src/traffic_server/InkAPI.cc

```diff
@@ -4510,7 +4519,16 @@ int
 TSContCall(TSCont contp, TSEvent event, void *edata)
 {
   Continuation *c = (Continuation *)contp;
-  return c->dispatchEvent((int)event, edata);
+  if (c->mutex != nullptr) {
+    MUTEX_TRY_LOCK(lock, c->mutex, this_ethread());
+    if (!lock.is_locked()) {
+      // If we cannot get the lock, the caller needs to restructure to handle rescheduling
+      ink_release_assert(0);
```

Comment:
The documentation also need to be changed.

> If there is a mutex an attempt will be made to lock that mutex. If either there is no mutex, or the mutex lock was acquired, the handler will be called directly. Otherwise there is a mutex and it was not successfully locked. In that case an event will be scheduled to dispatch as soon as possible, but not in the current call stack.

https://github.com/apache/trafficserver/blob/1ef8b2ad64f248db1a31d1eebaed4b011a4669b5/doc/developer-guide/api/functions/TSContCall.en.rst

Reply:
Interesting.  We are getting closer to the documentation anyway.  I'll update the docs here to reflect the functionality in this PR.  In the future it would be good to have a version of TSContCall that returns an action so that we can implement the try to get the mutex and schedule to perform later if needed, but return the action so the caller can handle the reschedule safely.

Reply:
Does the new version of the doc match reality?

Reply:
Yes, it looks good to me.